htop.c

```diff
@@ -137,9 +139,33 @@ static CommandLineSettings parseArguments(int argc, char** argv) {
             }
 
             while(pid) {
-                unsigned int num_pid = atoi(pid);
-                Hashtable_put(flags.pidWhiteList, num_pid, (void *) 1);
-                pid = strtok_r(NULL, ",", &saveptr);
+               char *endptr;
+               unsigned long num_pid;
+
+               if (pid[0] == '-') {
+                  fprintf(stderr, "Error: PID is negative \"%s\".\n", pid);
```

Comment:
The user wouldn't be able to see these messages since they would be quickly erased by the full screen interface. We can't expect users to pipe stderr just to catch errors.